src/components/composite/DepositFlow/DepositFlow.tsx

```diff
@@ -36,10 +66,24 @@ export const DepositFlowButton: FC<DepositFlowButtonProps> = ({
   ...props
 }) => {
   const { t } = useTranslation();
+  const { trackEvent } = useUserTracking();
   const openModal = useDepositFlowStore((state) => state.openModal);
+
+  const handleClick = () => {
+    trackEvent({
+      category: TrackingCategory.Earn,
+      action: TrackingAction.ClickEarnDepositButton,
+      label: 'click-earn-deposit-button',
+      data: {
+        [TrackingEventParameter.EarnOpportunitySlug]:
+          earnOpportunity.slug || '',
+      },
+    });
```

Comment:
It's leaking low level tracking implem, it'd be easier to maintain if with a few higher level functions (https://principles-wiki.net/principles:single_level_of_abstraction)


```suggestion
    trackEarnDeposit(earnOpportunity.slug)
```